apps/cowswap-frontend/src/common/hooks/orderProgressBarV2/useOrderProgressBarV2Props.ts

```diff
@@ -85,9 +85,15 @@ export function useOrderProgressBarV2Props(
     lastTimeChangedSteps,
     cancellationTriggered,
   } = useGetExecutingOrderState(orderId)
+
   const solversInfo = useSolversInfo(chainId)
   const totalSolvers = Object.keys(solversInfo).length
 
+  // When the order is in a final state or progress bar is disabled, avoid querying backend unnecessarily
+  const doNotQuery =
+    !!(order && getIsFinalizedOrder(order) && (order.status !== OrderStatus.FULFILLED || apiSolverCompetition)) ||
+    disableProgressBar
```

Comment:
looks ok, haven't tested, I struggle a bit to understand the condition though
Specially `(order.status !== OrderStatus.FULFILLED || apiSolverCompetition))` 

`FULFILLED` check comes after checking `isFinalizedOrder`

Wonder if it would help to make a function that encapsulates this check, and returns early in some cases offering a nice comment

Reply:
eb2bd5877c8a